### PR TITLE
refactor(timeutil): change deprecated TimeTools to TimeUtil

### PR DESCRIPTION
**Pull request by EikeSan**
Changed the use of TimeTools for TimeUtil

#225

### CODE REVIEW

**Comment by EikeSan** [comment]
Hello @ckittl  and @johanneshiry , i need some help with  the Jenkis CI, wich steps i need to follow to get it succeed?

File: src/main/java/edu/ie3/datamodel/io/factory/result/ConnectorResultFactory.java [hunk 1]
```diff
@@ -29,6 +30,8 @@
   private static final String ICMAG = "icmag";
   private static final String ICANG = "icang";
   private static final String TAPPOS = "tappos";
+  private TimeUtil timeUtil = TimeUtil.withDefaults;
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/io/factory/result/ConnectorResultFactory.java:33]:
```suggestion
  private final TimeUtil timeUtil = TimeUtil.withDefaults;
```

File: src/main/java/edu/ie3/datamodel/io/factory/result/NodeResultFactory.java [hunk 2]
```diff
@@ -8,17 +8,20 @@
...
 import javax.measure.quantity.Dimensionless;
-import tech.units.indriya.ComparableQuantity;
+import java.time.ZonedDateTime;
+import java.util.*;
 
 public class NodeResultFactory extends ResultEntityFactory<NodeResult> {
   private static final String VMAG = "vmag";
   private static final String VANG = "vang";
 
+  private TimeUtil timeUtil = TimeUtil.withDefaults;
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/io/factory/result/NodeResultFactory.java:23]:
```suggestion
  private final TimeUtil timeUtil = TimeUtil.withDefaults;
```

File: src/main/java/edu/ie3/datamodel/io/factory/result/ThermalResultFactory.java [hunk 3]
```diff
@@ -27,6 +27,8 @@
   private static final String ENERGY = "energy";
   private static final String FILL_LEVEL = "fillLevel";
 
+  private TimeUtil timeUtil = TimeUtil.withDefaults;
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/io/factory/result/ThermalResultFactory.java:30]:
```suggestion
  private final TimeUtil timeUtil = TimeUtil.withDefaults;
```

File: src/test/groovy/edu/ie3/datamodel/io/factory/result/ConnectorResultFactoryTest.groovy [hunk 4]
```diff
@@ -7,13 +7,18 @@ package edu.ie3.datamodel.io.factory.result
...
+import edu.ie3.datamodel.models.result.connector.Transformer2WResult
+import edu.ie3.datamodel.models.result.connector.Transformer3WResult
 import edu.ie3.test.helper.FactoryTestHelper
-import edu.ie3.util.TimeTools
+import edu.ie3.util.TimeUtil
 import spock.lang.Specification
 
 class ConnectorResultFactoryTest extends Specification implements FactoryTestHelper {
 
+	private final TimeUtil timeUtil = TimeUtil.withDefaults
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/io/factory/result/ConnectorResultFactoryTest.groovy:20]: I would like you to move this object to `FactoryTestHelper` (as a static member). Then you can avoid instantiating it in every factory test.

File: src/test/groovy/edu/ie3/datamodel/io/factory/result/NodeResultFactoryTest.groovy [hunk 5]
```diff
@@ -10,11 +10,13 @@ import edu.ie3.datamodel.io.factory.SimpleEntityData
 import edu.ie3.datamodel.models.StandardUnits
 import edu.ie3.datamodel.models.result.NodeResult
 import edu.ie3.test.helper.FactoryTestHelper
-import edu.ie3.util.TimeTools
+import edu.ie3.util.TimeUtil
 import spock.lang.Specification
 
 class NodeResultFactoryTest extends Specification implements FactoryTestHelper {
 
+	private final TimeUtil timeUtil = TimeUtil.withDefaults
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/io/factory/result/NodeResultFactoryTest.groovy:18]: See above.

File: src/test/groovy/edu/ie3/datamodel/io/factory/result/SystemParticipantResultFactoryTest.groovy [hunk 6]
```diff
@@ -10,12 +10,14 @@ import edu.ie3.datamodel.io.factory.SimpleEntityData
...
 import edu.ie3.datamodel.models.result.system.*
 import edu.ie3.test.helper.FactoryTestHelper
-import edu.ie3.util.TimeTools
+import edu.ie3.util.TimeUtil
 import spock.lang.Specification
 import tech.units.indriya.unit.Units
 
 class SystemParticipantResultFactoryTest extends Specification implements FactoryTestHelper {
 
+	private final TimeUtil timeUtil = TimeUtil.withDefaults
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/io/factory/result/SystemParticipantResultFactoryTest.groovy:19]: See above.

File: src/test/groovy/edu/ie3/datamodel/io/factory/result/ThermalResultFactoryTest.groovy [hunk 7]
```diff
@@ -10,12 +10,14 @@ import edu.ie3.datamodel.models.StandardUnits
...
 import edu.ie3.datamodel.models.result.thermal.ThermalHouseResult
 import edu.ie3.datamodel.models.result.thermal.ThermalUnitResult
-import edu.ie3.util.TimeTools
+import edu.ie3.util.TimeUtil
 import spock.lang.Specification
 import tech.units.indriya.quantity.Quantities
 
 class ThermalResultFactoryTest extends Specification {
 
+	private final TimeUtil timeUtil = TimeUtil.withDefaults
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/io/factory/result/ThermalResultFactoryTest.groovy:19]: See above.

**Review by ckittl — CHANGES_REQUESTED** [review]
Thanks a lot @EikeSan for your contributions! 🥳 
I have had a quick look at your changes and they mostly look great! I only would wish for some slight adaptions in order to meet our (quite eager) code quality ambitions. 😉

**Comment by ckittl** [comment]
> Hello @ckittl and @johanneshiry , i need some help with the Jenkis CI, wich steps i need to follow to get it succeed?

Seems to be some issue, that _might_ come from our CI. @johanneshiry is our CI expert, but I will try and have a look at it as well. :-)

**Comment by ckittl** [comment]
> Hello @ckittl and @johanneshiry , i need some help with the Jenkis CI, wich steps i need to follow to get it succeed?

Okay, I think I have figured you, what's happening. Our CI has the capability to check, if the branch to be merged in properly follows semantic versioning. It recognizes different merge cases (from _dev_ to _main_ or from a feature branch to _dev_, ...) based on the branch naming. When designing the script, we did not thoroughly consider merging from a fork. This case (merging from _dev_ into _dev_) is not covered, yet. This is why the CI is failing.

The easiest way to overcome this, would be kindly asking you to establish a new branch in your fork (e.g. _es/#225-timeUtil_) and open a new PR (I think changing the source from where to merge is not possible, isn't it?). The other would be to wait for @johanneshiry. But I'm not sure, if those adoptions would meet his (timely) resources right now.

**Comment by EikeSan** [comment]
@ckittl I only can change de base branch, i will close this  and open a new one!

**Pull request closed without merging**